pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/TriggersTest.java

```diff
@@ -162,4 +176,113 @@ public void actualTriggerCorrectScope() throws Exception {
         expect("actualTriggerCorrectScope")
                 .go();
     }
+
+    @LocalData
+    @Test
+    public void doNotRestartEqualTriggers() throws Exception {
```

Comment:
This is good. 
We also need a test to show this with two triggers and that if the order of the triggers changes (but they are otherwise the same) that this still works and does not recreate the triggers.